src/Microsoft.Identity.Web/WebAppAuthenticationBuilderExtensions.cs

```diff
@@ -100,101 +95,168 @@ public static partial class WebAppAuthenticationBuilderExtensions
+                           OpenIdConnectParameterNames.DomainHint);
+
+                       // delete the login_hint and domainHint from the Properties when we are done otherwise
+                       // it will take up extra space in the cookie.
+                       context.Properties.Parameters.Remove(OpenIdConnectParameterNames.LoginHint);
+                       context.Properties.Parameters.Remove(OpenIdConnectParameterNames.DomainHint);
+                   }
+
+                   context.ProtocolMessage.SetParameter(Constants.ClientInfo, Constants.One);
+
+                   // Additional claims
+                   if (context.Properties.Items.ContainsKey(OidcConstants.AdditionalClaims))
+                   {
+                       context.ProtocolMessage.SetParameter(
+                           OidcConstants.AdditionalClaims,
+                           context.Properties.Items[OidcConstants.AdditionalClaims]);
+                   }
+
+                   if (microsoftIdentityOptions.Value.IsB2C)
+                   {
+                       // When a new Challenge is returned using any B2C user flow different than susi, we must change
+                       // the ProtocolMessage.IssuerAddress to the desired user flow otherwise the redirect would use the susi user flow
+                       await b2cOidcHandlers.OnRedirectToIdentityProvider(context).ConfigureAwait(false);
+                   }
+
+                   await redirectToIdpHandler(context).ConfigureAwait(false);
+               };
+
+               if (microsoftIdentityOptions.Value.IsB2C)
+               {
+                   var remoteFailureHandler = options.Events.OnRemoteFailure;
+                   options.Events.OnRemoteFailure = async context =>
+                   {
+                       // Handles the error when a user cancels an action on the Azure Active Directory B2C UI.
+                       // Handle the error code that Azure Active Directory B2C throws when trying to reset a password from the login page
+                       // because password reset is not supported by a "sign-up or sign-in user flow".
+                       await b2cOidcHandlers.OnRemoteFailure(context).ConfigureAwait(false);
+
+                       await remoteFailureHandler(context).ConfigureAwait(false);
+                   };
+               }
+
+               if (subscribeToOpenIdConnectMiddlewareDiagnosticsEvents)
+               {
+                   var diags = serviceProvider.GetRequiredService<IOpenIdConnectMiddlewareDiagnostics>();
+
+                   diags.Subscribe(options.Events);
+               }
+           });
 
             return builder;
         }
+
+        internal static void PopulateOpenIdOptionsFromMicrosoftIdentityOptions(OpenIdConnectOptions options, MicrosoftIdentityOptions microsoftIdentityOptions)
+        {
+            options.Authority = microsoftIdentityOptions.Authority;
```

Comment:
Seems a bit too manual to copy each property but, I guess, based on Chris's [comment](https://github.com/AzureAD/microsoft-identity-web/issues/123#issuecomment-648925848) that's how we have to do...

Reply:
@pmaytak @jennyf19 
I've created a unit test (see below) to check if the properties get change in future versions of ASP.NET Core